File: LeetCode/Python/#53 Maximum Subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums)==1:return nums[0]

        this_sum = max_sum = nums[0]
        for i in nums[1:]:
            this_sum = max(i,this_sum+i)
            max_sum = max(max_sum,this_sum)

        return max_sum

File: LeetCode/JavaScript/#53 Maximum Subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
	if(nums.length===1) return nums[0];
    var sum = nums[0];
    var this_sum = nums[0];
    var len = nums.length;
    for(let i =1;i<len;i++){
    	this_sum =Math.max(nums[i],this_sum+nums[i]);
    	sum = Math.max(this_sum,sum);
    }
    return sum;
};